ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -231,6 +268,57 @@ private HelixBootstrapUpgradeUtil(String hardwareLayoutPath, String partitionLay
     }
   }
 
+  /**
+   * Create HelixPropertyStore for all datacenters. Also it checks if the zNode exists, if not the method will create a
+   * new zNode for PartitionOverride.
+   */
+  private void createHelixPropertyStoreForAllDCs() {
+    Properties storeProps = new Properties();
+    storeProps.setProperty("helix.property.store.root.path", "/" + this.clusterName);
+    HelixPropertyStoreConfig propertyStoreConfig = new HelixPropertyStoreConfig(new VerifiableProperties(storeProps));
+    for (Map.Entry<String, ClusterMapUtils.DcZkInfo> entry : dataCenterToZkAddress.entrySet()) {
+      ZkClient zkClient = new ZkClient(entry.getValue().getZkConnectStr(), propertyStoreConfig.zkClientSessionTimeoutMs,
+          propertyStoreConfig.zkClientConnectionTimeoutMs, new ZNRecordSerializer());
+      List<String> subscribedPaths = Collections.singletonList(propertyStoreConfig.rootPath);
+      HelixPropertyStore<ZNRecord> helixPropertyStore =
+          new ZkHelixPropertyStore<>(new ZkBaseDataAccessor<>(zkClient), propertyStoreConfig.rootPath, subscribedPaths);
+      if (!helixPropertyStore.exists(PROPERTYSTORE_PATH, AccessOption.PERSISTENT)) {
+        ZNRecord znRecord = new ZNRecord(ZNRECORD_NAME);
+        helixPropertyStore.create(PROPERTYSTORE_PATH + "/" + ZNRECORD_NAME, znRecord, AccessOption.PERSISTENT);
+        info("Creating a new zNode: ", PROPERTYSTORE_PATH + "/" + ZNRECORD_NAME);
+      }
+      dataCenterToPropertyStore.put(entry.getKey(), helixPropertyStore);
+    }
+  }
+
+  /**
+   * Uploads the seal state of all partitions in the format of map.
+   * TODO: add partitionClass info of all partitions into map.
+   */
+  private void uploadPartitionOverride() {
+    Map<String, Map<String, String>> partitionOverrideInfos = new HashMap<>();
+    for (PartitionId partitionId : staticClusterMap.getAllPartitionIds(null)) {
+      String partitionName = partitionId.toPathString();
+      Map<String, String> partitionProperties = new HashMap<>();
+      partitionProperties.put("state", "RW");
+      for (ReplicaId replicaId : partitionId.getReplicaIds()) {
+        if (replicaId.isSealed()) {
+          partitionProperties.put("state", "RO");
+          break;
+        }
+      }
+      partitionOverrideInfos.put(partitionName, partitionProperties);
+    }
+    info("Setting partition override for all datacenters.");
+    // set partitionOverride in PROPERTYSTORE for all datacenters
+    for (HelixPropertyStore<ZNRecord> helixPropertyStore : dataCenterToPropertyStore.values()) {
+      ZNRecord znRecord = new ZNRecord(ZNRECORD_NAME);
+      znRecord.setMapFields(partitionOverrideInfos);
+      String path = "/" + this.clusterName + PROPERTYSTORE_PATH + "/" + ZNRECORD_NAME;
+      helixPropertyStore.set(path, znRecord, AccessOption.PERSISTENT);
```

Comment:
Check the return value of `set`. 

Also, you may not need to create the znode ahead of time in `createHelixPropertyStoreForAllDCs()` since the `set` method will take care of that. In fact, if you do this, you would not need `createHelixPropertyStoreForAllDCs()` and you could just instantiate the zk client within `uploadPartitionOverride`

Reply:
sounds reasonable. I will make it more concise.